mesonbuild/options.py

```diff
@@ -93,16 +97,23 @@ class OptionKey:
     __slots__ = ['name', 'subproject', 'machine', '_hash']
 
     name: str
-    subproject: str
+    subproject: T.Optional[str] # None is global, empty string means top level project
```

Comment:
I agree, but when I was working on this I had to aggressively descope fixes and changes just to get the core change done. These are things that can be changed later. And in case later never comes (which sadly can happen), at least this isolated part works